cura/CuraPackageManager.py

```diff
@@ -0,0 +1,357 @@
+        self._to_install_package_dict = {}  # a dict of packages that need to be installed at the next start
+
+    installedPackagesChanged = pyqtSignal()  # Emitted whenever the installed packages collection have been changed.
+
+    def initialize(self):
+        self._loadManagementData()
+        self._removeAllScheduledPackages()
+        self._installAllScheduledPackages()
+
+    # (for initialize) Loads the package management file if exists
+    def _loadManagementData(self) -> None:
+        if not os.path.exists(self._package_management_file_path):
+            Logger.log("i", "Package management file %s doesn't exist, do nothing", self._package_management_file_path)
+            return
+
+        with open(self._package_management_file_path, "r", encoding = "utf-8") as f:
+            management_dict = json.loads(f.read(), encoding = "utf-8")
+
+            self._installed_package_dict = management_dict["installed"]
+            self._to_remove_package_set = set(management_dict["to_remove"])
+            self._to_install_package_dict = management_dict["to_install"]
+
+            Logger.log("i", "Package management file %s is loaded", self._package_management_file_path)
+
+    def _saveManagementData(self) -> None:
+        with open(self._package_management_file_path, "w", encoding = "utf-8") as f:
+            data_dict = {"installed": self._installed_package_dict,
+                         "to_remove": list(self._to_remove_package_set),
+                         "to_install": self._to_install_package_dict}
+            data_dict["to_remove"] = list(data_dict["to_remove"])
+            json.dump(data_dict, f)
+            Logger.log("i", "Package management file %s is saved", self._package_management_file_path)
+
+    # (for initialize) Removes all packages that have been scheduled to be removed.
+    def _removeAllScheduledPackages(self) -> None:
+        print("Will purge", self._to_remove_package_set)
+        for package_id in self._to_remove_package_set:
+            self._purgePackage(package_id)
+        self._to_remove_package_set.clear()
+        self._saveManagementData()
+
+    # (for initialize) Installs all packages that have been scheduled to be installed.
+    def _installAllScheduledPackages(self) -> None:
+        for package_id, installation_package_data in self._to_install_package_dict.items():
+            self._installPackage(installation_package_data)
+        self._to_install_package_dict.clear()
+        self._saveManagementData()
+
+    # Checks the given package is installed. If so, return a dictionary that contains the package's information.
+    def getInstalledPackageInfo(self, package_id: str) -> Optional[dict]:
+        if package_id in self._to_remove_package_set:
+            return None
+
+        if package_id in self._to_install_package_dict:
+            package_info = self._to_install_package_dict[package_id]["package_info"]
+            package_info["is_bundled"] = False
```

Comment:
So this `getInstalledPackageInfo` getter changes the metadata of a package. I think that is unexpected, right?